pkg/beholder/client.go

```diff
@@ -346,6 +357,27 @@ func (c Client) ForName(name string) Client {
 	return newClient
 }
 
+// SetSigner updates the signer in the lazy signer if one was configured during client initialization.
+// This method enables setting the signer after the beholder client has been created, which is useful
+// when the signer is not available at client initialization time but the client needs to be configured
+// with rotating auth.
+func (c *Client) SetSigner(signer Signer) error {
+	if c.lazySigner == nil {
+		return fmt.Errorf("no lazy signer configured - client was not initialized with a LazySigner")
+	}
+	c.lazySigner.SetSigner(signer)
+	return nil
+}
+
+// HasSigner returns true if a signer has been set in the lazy signer.
+// Returns false if no lazy signer was configured or if the keystore has not been set yet.
+func (c *Client) HasSigner() bool {
+	if c.lazySigner == nil {
+		return false
+	}
+	return c.lazySigner.HasSigner()
+}
```

Comment:
We could eliminate the possibility of nil by embedding the `lazySigner struct` directly rather than the `LazySigner interface`.
```suggestion
func (c *Client) SetSigner(signer Signer) {
	c.lazySigner.SetSigner(signer)
}

// HasSigner returns true if a signer has been set in the lazy signer.
// Returns false if no lazy signer was configured or if the keystore has not been set yet.
func (c *Client) HasSigner() bool {
	return c.lazySigner.HasSigner()
}
```

Reply:
Can't embed as value - rotatingAuth and Client need to share the same lazySigner instance. If we embed it, they'd each get a copy and SetSigner() would only update the client's copy, not the one rotatingAuth is using.